lib/virtio/virtqueue.c

```diff
@@ -632,6 +733,13 @@ static int vq_ring_must_notify(struct virtqueue *vq)
 {
 	uint16_t new_idx, prev_idx, event_idx;
 
+#ifdef VIRTIO_D_CACHE
+	metal_cache_invalidate(VRING_USED(vq), vring_used_size(VRING_NUM(vq)));
+	metal_cache_invalidate(VRING_AVAIL(vq),
+			       vring_avail_size(VRING_NUM(vq),
```

Comment:
only master should flush this

Reply:
assuming you mean invalidate avail could have been modified by master in `vq_ring_enable_interrupt()` so we need to invalidate this also for slave